02-microarray/dimension-reduction_microarray_01_pca.Rmd

```diff
@@ -0,0 +1,312 @@
+
+Attach the packages we need for this analysis:
+
+```{r}
+# Attach the library
+library(ggplot2)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+
+# Set the seed so our results are reproducible:
+set.seed(12345)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE37382.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE37382.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene id column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+```
+
+
+<!--ALL THE ANALYSIS STEPS HERE -->
+
+Now we are going to use a combination of functions from base R and the `ggplot2` package to perform and visualize the results of the Principal Component Analysis (PCA) dimension reduction technique on our medulloblastoma samples.
+
+## Perform Principal Components Analysis
+
+In this code chunk, we are going to perform Principal Component Analysis (PCA) on our data and create a data frame using the PCA scores and the variables from our metadata that we are going to use to annotate our plot later.
+We are using the base R [`prcomp()`](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/prcomp) function to perform Principal Component Analysis (PCA) here [@prcomp-function].
+
+```{r}
+# Perform Principal Component Analysis (PCA)
+pca <- prcomp(t(df)) # We have to transpose our data frame to meet the `prcomp()` function requirements
+
+# Make into data frame for plotting PC1 and PC2 with `ggplot2`
+pca_df <- data.frame(pca$x[, 1:2]) %>%
+  # Turn samples_ids stored as rownames into column
+  tibble::rownames_to_column("refinebio_accession_code") %>%
+  # Bring only the variables that we want from the metadata into this data frame -- here we are going to join by `refinebio_accession_code` values
+  dplyr::inner_join(dplyr::select(
+    metadata, refinebio_accession_code,
+    histology, subgroup
+  ),
+  by = "refinebio_accession_code"
+  )
+```
+
+Let's plot the PC scores for the first two principal components.
```

Comment:
I might add some steps in here that include variance explained on the plot. You might find this portion of training material to be helpful in making those changes: https://github.com/AlexsLemonade/training-modules/blob/3dbc6f3f53c680ec6aa2f513851c1cd4635cc31c/machine-learning/02-openpbta_consensus_clustering.Rmd#L310